include/rmm/mr/device/detail/stream_ordered_memory_resource.hpp

```diff
@@ -356,29 +362,46 @@ class stream_ordered_memory_resource : public crtp<PoolResource>, public device_
    * @return A block with non-null pointer and size >= `size`, or a nullptr block if none is
    *         available in `blocks`.
    */
-  block_type get_block_from_other_stream(size_t size, stream_event_pair stream_event)
+  block_type get_block_from_other_stream(size_t size,
+                                         stream_event_pair stream_event,
+                                         free_list& blocks,
+                                         bool merge_first)
```

Comment:
Style-wise I would prefer it as a scoped enumerator but I don't feel strongly. Leaving it as a `bool` is ok as well.

Reply:
Since it's all very localized and internal to the class I prefer to leave it.